xwayland/xwayland.c

```diff
@@ -123,9 +123,23 @@ static void exec_xwayland(struct wlr_xwayland *wlr_xwayland) {
 		wlr_xwayland->wl_fd[1], wlr_xwayland->display, wlr_xwayland->x_fd[0],
 		wlr_xwayland->x_fd[1], wlr_xwayland->wm_fd[1]);
 
-	// TODO: close stdout/err depending on log level
+	// Closes stdout/stderr depending on log verbosity
+	enum wlr_log_importance verbosity = wlr_log_get_verbosity();
+	int devnull = open("/dev/null", O_WRONLY | O_CREAT, 0666);
+	if (devnull < 0) {
+		wlr_log_errno(WLR_ERROR, "XWayland: failed to open /dev/null");
+		_exit(EXIT_FAILURE);
+	}
+	if (verbosity < WLR_INFO) {
+		dup2(devnull, STDOUT_FILENO);
+	}
+	if (verbosity < WLR_ERROR) {
+		dup2(devnull, STDERR_FILENO);
+	}
 
 	execvp("Xwayland", argv);
+
+	close(devnull);
```

Comment:
This is never reached. We should probably close before exec-ing?

Reply:
It will be reached on failure, but there deserves to be some proper error handling here.

Reply:
You mean checking the result of execvp, right?

Reply:
Actually, looking at how exec_wayland is used, there is already error handling outside of the function.
Closing devnull here will mess with errno, though.

Reply:
The error checking code in `xwayland_start_server` (line 349) could probably be shifted to after the `execvp` call (since execvp returns iff it fails)?

This in particular:
```
wlr_log_errno(WLR_ERROR, "failed to exec Xwayland");
_exit(EXIT_FAILURE);
```

Should I add it to the PR?

Reply:
Yeah, you may as well. Also mark `exec_wayland` as [noreturn](https://en.cppreference.com/w/c/language/_Noreturn).

Reply:
Added to commit 8184ccc.